### DISCUSSION TITLE
UTF-8 token input mismatch.

### DISCUSSION THREAD

**Discussion by vega0** [antlr/antlr4#4260] [Q&A]
https://i.imgur.com/kYgznyf.png

```
TOKEN_BULEVA_ZNACHENIYA_ISTINY: '\u0418\u0441\u0442\u0438\u043D\u0430'; // Истина
TOKEN_BULEVA_ZNACHENIYA_LOJI: '\u041B\u043E\u0436\u044C'; // Ложь.

bulevoZnacheniye: TOKEN_BULEVA_ZNACHENIYA_LOJI | TOKEN_BULEVA_ZNACHENIYA_ISTINY;
```

By some reason UTF-8 input token mismatch.

**Comment by jimidle** [comment]
You’re lexer rules are in conflict. Are you sure you don’t need fragments?
Lexer rules are not driven by the parser.

**Reply by vega0** [reply]
Can you tell me then the best way to make utf8 constants keyword matching?

**Reply by vega0** [reply]
fragments did not work .
